task-driver/integration/tests/update_wallet.rs

```diff
@@ -47,57 +71,233 @@ async fn test_update_wallet_then_recover(test_args: IntegrationTestArgs) -> Resu
+    )
+    .await
+}
+integration_test_async!(test_update_wallet__add_fee);
+
+/// Tests updating a wallet by removing a fee from the wallet
+#[allow(non_snake_case)]
+async fn test_update_wallet__remove_fee(test_args: IntegrationTestArgs) -> Result<()> {
+    let client = &test_args.starknet_client;
+    let mut rng = thread_rng();
+
+    // Create a new wallet with a non-empty fee and post it on-chain
     let blinder_seed = Scalar::random(&mut rng);
     let share_seed = Scalar::random(&mut rng);
     let mut wallet = empty_wallet_from_seed(blinder_seed, share_seed);
 
+    wallet.fees.push(DUMMY_FEE.clone());
+
     allocate_wallet_in_darkpool(&wallet, client).await?;
 
-    // Update the wallet by inserting an order
+    // Update the wallet by removing a fee
     let old_wallet = wallet.clone();
-    wallet.orders.insert(
-        Uuid::new_v4(),
-        Order {
-            quote_mint: 0u8.into(),
-            base_mint: 1u8.into(),
-            side: OrderSide::Buy,
-            amount: 10,
-            worst_case_price: FixedPoint::from_integer(10),
-            timestamp: get_current_time_seconds(),
+    wallet.fees.pop();
+    wallet.reblind_wallet();
+
+    execute_wallet_update_and_verify(
+        old_wallet,
+        wallet,
+        None, /* transfer */
+        blinder_seed,
+        share_seed,
+        test_args,
+    )
+    .await
+}
+integration_test_async!(test_update_wallet__remove_fee);
+
+// ------------
+// | Balances |
+// ------------
+
+/// Tests updating a wallet by depositing into the pool
+#[allow(non_snake_case)]
+async fn test_update_wallet__deposit_and_withdraw(test_args: IntegrationTestArgs) -> Result<()> {
+    let client = &test_args.starknet_client;
+
+    // Create a new wallet and post it on-chain
+    let (mut wallet, blinder_seed, share_seed) = new_wallet_in_darkpool(client).await?;
+
+    // Update the wallet by depositing into the pool
+    let old_wallet = wallet.clone();
+
+    let mint = biguint_from_hex_string(&test_args.erc20_addr);
+    let amount = 10u64;
+
+    wallet.balances.insert(
+        mint.clone(),
+        Balance {
+            mint: mint.clone(),
+            amount,
         },
     );
     wallet.reblind_wallet();
 
-    let task = UpdateWalletTask::new(
-        get_current_time_seconds(),
-        None, /* external_transfer */
+    // Approve the deposit on the ERC20 contract
+    increase_erc20_allowance(amount, test_args.clone()).await?;
+
+    let account_addr = biguint_from_hex_string(&test_args.account_addr);
+    execute_wallet_update_and_verify(
         old_wallet,
         wallet.clone(),
-        client.clone(),
-        test_args.network_sender.clone(),
-        test_args.global_state.clone(),
-        test_args.proof_job_queue.clone(),
-    )?;
+        Some(ExternalTransfer {
+            mint: mint.clone(),
+            amount: amount.into(),
+            account_addr: account_addr.clone(),
+            direction: ExternalTransferDirection::Deposit,
```

Comment:
thoughts on making this a `lazy_static` item as well? i.e. a `DUMMY_DEPOSIT` and a `DUMMY_WITHDRAWAL`?

Reply:
It's parameterized by a CLI arg, so it isn't static